src/airbrakes.py

```diff
@@ -1,6 +1,181 @@
+step_angle: Step angle in degrees of stepper motor
+microsteps: 2 (half), 4 (quarter), 16 (sixteenth) etc.
+gear_ratio: (# of motor turns) / (# of gearbox output turns)
+
+__max_steps_to_open: Maximum numper of steps needed to fully open the brakes.
+This is calculated from the motor parameters: step_angle, microsteps, gear_ratio
+
+
+Methods
+-------
+
+__singleStep():
+  Steps Airbrakes stepper motor once in specified direction.
+  Returns nothing.
+
+  Param: step_direction
+    Passing step_direction=True opens Airbrakes one step
+    Passing step_direction=False closes Airbrakes one step
+
+    Note: If passing True closes Airbrakes, invert the value of 
+    step_direction when initializing Airbrakes object
+
+calibrate(): 
+  Sets __max_pot_val by reading potentiometer in the fully open state
+  and sets __min_pot_val by reading the potentiometer in the closed state
+  Returns nothing.
+
+  Must be called before launch (on the launch pad) to calibrate the system.
+
+deployBrakes():
+  Opens the brakes to the specifed percentage of fully open.
+
+  Note: This uses the potentiometer for feedback so there will be a small
+  error associated with this. Adjust this error by changing max_error in
+  method body.
+
+  Returns value of potentiometer at final position.
+
+"""
+class Airbrakes:
+  
+  # Set when calibrate() is called
+  __max_pot_val = None
+  __min_pot_val = None
+
+  def __init__(self,  stepper_pin, direction_pin, direction: bool,
+               ads_pot_pin=0, step_angle=1.8, microsteps=16,
+               step_delay_micro_sec=75, gear_box_ratio=14):
+
+    # GPIO pins
+    self.step_pin = stepper_pin
+    self.dir_pin  = direction_pin
+
+    # Stepper motor specs
+    self.motor_direction: bool = direction
+    self.step_delay = step_delay_micro_sec/1000000.0
+    self.step_angle = step_angle
+    self.microsteps = microsteps
+    self.gear_ratio = gear_box_ratio
+   
+    # Max steps to go from fully closed to fully open with 1.25 FoS
+    # Division by 6 is because airbrakes only require 1/6 of a turn to open
+    self.__max_steps_to_open = 1.25*(self.gear_ratio*(360/self.step_angle)*self.microsteps)/6
+
+    # Use numbering that appears on PCB
+    GPIO.setmode(GPIO.BOARD)
+
+    # Set the pins as outputs
+    GPIO.setup(self.step_pin, GPIO.OUT)
+    GPIO.setup(self.dir_pin, GPIO.OUT)
+
+    # Initialize ADC (ADS1115)
+    self.i2c = busio.I2C(board.SCL, board.SDA)
+    self.ads = ADS.ADS1115(self.i2c)
+    self.potentiometer = AnalogIn(self.ads, ads_pot_pin)
+
+  def __singleStep(self, step_direction: bool):
+
+    # ^ XORs the direction (inverting it if motor_direction dictates it)
+    GPIO.output(self.dir_pin, step_direction ^ self.motor_direction)
+
+    # Pulse step pin
+    GPIO.output(self.step_pin, GPIO.HIGH)
+    sleep(self.step_delay)
+    GPIO.output(self.step_pin, GPIO.LOW)
+
+  def calibrate(self):
+    
+    # Fully open brakes and record pot value
+    for _ in range(self.__max_steps_to_open):
```

Comment:
yo what is this _ syntax??? this is a wild iterator name

Reply:
Ah thats from me writing too much MATLAB. Basically that IDE complains if you don't use the iterator, but you can suppress the warning by naming it `_`. I did that here since I don't actually use the iterator, I just need to loop to run a certain number of times.

Reply:
I can change it if you like, but it probably doesn't matter?

Reply:
Nah thats all good since its not used in the loop